### ISSUE TITLE
False positive in redos detection when nested quantifier mutually exclusive

### ISSUE THREAD

**Issue by mschwager** [duo-labs/dlint#45]
"A group that contains a token with a quantifier must not have a quantifier of its own unless the quantified token inside the group can only be matched with something else that is mutually exclusive with it." ([Nested Quantifiers](https://www.regular-expressions.info/redos.html))

Dlint does not currently eliminate safe regular expressions that have nested quantifiers but they're mutually exclusive. Consider the example from the above link:

```
$ python -m dlint.redos -p '(x\w{1,10})+y'
('(x\\w{1,10})+y', True)
```

Dlint finds the nested quantifier. But it flags the corrected code as well:

```
$ python -m dlint.redos -p '(x[a-wyz0-9_]{1,10})+y'
('(x[a-wyz0-9_]{1,10})+y', True)
```

This example is okay because there's no character overlap inside the nested quantifier. We should fix this false positive.
